### PR TITLE
Fixed #26144 -- Added warning when dumping proxy model

### PR DESCRIPTION
**Pull request by yoongkang**

### CODE REVIEW

**Comment by timgraham** [comment]
Please see the ticket for feedback on the implementation.

**Comment by yoongkang** [comment]
Does this look okay?

File: django/core/management/commands/dumpdata.py [hunk 1]
```diff
@@ -132,9 +138,15 @@ def get_objects(count_only=False):
...
             """
-            for model in serializers.sort_dependencies(app_list.items()):
+            models = serializers.sort_dependencies(app_list.items())
+            for model in models:
                 if model in excluded_models:
                     continue
+                if model._meta.proxy and inspect.getmro(model)[1] not in models:
+                    warnings.warn(
+                        "%s is a Proxy model. It won't be serialized." %
+                        model.__name__,
```
Comment (timgraham) [inline django/core/management/commands/dumpdata.py]: I'd use `model._meta.label` (which includes the `app_label`).

File: django/core/management/commands/dumpdata.py [hunk 2]
```diff
@@ -132,9 +138,15 @@ def get_objects(count_only=False):
...
+                        category=ProxyModelWarning)
```
Comment (timgraham) [inline django/core/management/commands/dumpdata.py]: We usually put the closing parenthesis on the next line.

File: django/core/management/commands/dumpdata.py [hunk 3]
```diff
@@ -132,9 +138,15 @@ def get_objects(count_only=False):
...
             count the number of objects to be serialized.
             """
-            for model in serializers.sort_dependencies(app_list.items()):
+            models = serializers.sort_dependencies(app_list.items())
+            for model in models:
                 if model in excluded_models:
                     continue
+                if model._meta.proxy and inspect.getmro(model)[1] not in models:
+                    warnings.warn(
+                        "%s is a Proxy model. It won't be serialized." %
```
Comment (timgraham) [inline django/core/management/commands/dumpdata.py]: proxy model and won't ...

File: tests/fixtures/tests.py [hunk 4]
```diff
@@ -476,6 +478,32 @@ def test_dumpdata_progressbar(self):
             management.call_command('dumpdata', 'fixtures', **options)
             self.assertEqual(new_io.getvalue(), '')
 
+    def test_dumpdata_proxy_without_concrete(self):
+        ProxySpy.objects.create(name='Paul')
+
+        with warnings.catch_warnings(record=True) as warning_list:
+            warnings.simplefilter("always")
+            self._dumpdata_assert(
```
Comment (timgraham) [inline tests/fixtures/tests.py]: single line?

File: tests/fixtures/tests.py [hunk 5]
```diff
@@ -476,6 +478,32 @@ def test_dumpdata_progressbar(self):
             management.call_command('dumpdata', 'fixtures', **options)
             self.assertEqual(new_io.getvalue(), '')
 
+    def test_dumpdata_proxy_without_concrete(self):
+        ProxySpy.objects.create(name='Paul')
+
+        with warnings.catch_warnings(record=True) as warning_list:
+            warnings.simplefilter("always")
```
Comment (timgraham) [inline tests/fixtures/tests.py]: single quotes for consistency

File: tests/fixtures/tests.py [hunk 6]
```diff
@@ -476,6 +478,32 @@ def test_dumpdata_progressbar(self):
...
+            self._dumpdata_assert(
+                ['fixtures.ProxySpy'],
+                '[]'
+            )
+            warning = warning_list.pop()
```
Comment (timgraham) [inline tests/fixtures/tests.py]: deindent

File: tests/fixtures/tests.py [hunk 7]
```diff
@@ -476,6 +478,32 @@ def test_dumpdata_progressbar(self):
...
+            self.assertEqual(warning.category, ProxyModelWarning)
+            self.assertEqual(
+                str(warning.message),
+                "ProxySpy is a Proxy model. It won't be serialized.")
```
Comment (timgraham) [inline tests/fixtures/tests.py]: parenthesis to next line

**Comment by yoongkang** [comment]
@timgraham Thanks for the review. I've made the changes.

**Comment by timgraham** [comment]
merged in 0edb8a146fd6c60f4c3b1ad0a4a89963962f22af, thanks!

**Pull request closed without merging**